Communications/Client.cs

```diff
@@ -51,20 +49,23 @@ public static class Client
 
         public static string Name { get; private set; } // Name of the client.
         public static bool IsConnected { get; private set; } // Whether or not the client and server are connected
-        public static List<Packet> SentPackets { get; private set; } // Storage for send packets.
-        public static List<Packet> ReceivedPackets { get; private set; } // Storage for received packets.
         public static event EventHandler<ConnectionStatusChanged> ClientConnectionChanged; // Invoked when the Client detects a connection change
+
         public static bool StorePackets; // Whether or not the client stores packets
+        public static List<Packet> SentPackets { get; private set; } // Storage for send packets.
+        public static List<Packet> ReceivedPackets { get; private set; } // Storage for received packets.
 
         /// <summary>
         /// Starts a Client process.
         /// </summary>
-        /// <param name="ServerIP">String representation of the IP Address of server.</param>
-        /// <param name="PortTCP">Target port for TCP Communications on the server.</param>
-        /// <param name="PortUDP">Target port for UDP Communications on the server.</param>
-        /// <param name="ReceiveBufferSize">Size of buffer for incoming data.</param>
-        /// <param name="OperationPeriod">Time in between receiving and sending individual packets.</param>
-        public static void Start(string ServerIP, int PortTCP, int PortUDP, string Name, int ReceiveBufferSize = 64, int OperationPeriod = 20)
+        /// <param name="ServerIP"> String representation of the IP Address of server.</param>
+        /// <param name="PortTCP"> Target port for TCP Communications on the server.</param>
+        /// <param name="PortUDP"> Target port for UDP Communications on the server.</param>
+        /// <param name="Name"> Name of client. </param>
+        /// <param name="UsePriorityQueue"> If it is true, use `GenericController` as sending buffer. Use `QueueBuffer` otherwise. </param>
+        /// <param name="ReceiveBufferSize"> Size of buffer for incoming packet. Unit: byte </param>
+        /// <param name="OperationPeriod"> Time to wait after receiving or sending each packet. Unit: ms </param>
+        public static void Start(string ServerIP, int PortTCP, int PortUDP, string Name, bool UsePriorityQueue = true, int ReceiveBufferSize = 64, int OperationPeriod = 20)
```

Comment:
Please set `UsePriorityQueue` to false by default. Since most subsystems are not going to need priority-based sending, it makes more sense to turn it off by default, only enabling it when they do need it.

Reply:
Also, can we make it the last argument in the method footprint?